File: src/codegeass/dashboard/models/hook.py
"""Hook models for API."""

from typing import Any

from pydantic import BaseModel, Field


class HookSummary(BaseModel):
    """Summary view of a hook tag."""

    tag: str
    description: str
    location: str  # "project" or "global"
    events: list[str] = Field(default_factory=list)


class Hook(BaseModel):
    """Full hook model."""

    tag: str
    description: str = ""
    hooks: dict[str, list[dict[str, Any]]] = Field(default_factory=dict)
    location: str | None = None  # "project" or "global"


class HookCreate(BaseModel):
    """Model for creating a new hook."""

    tag: str
    description: str = ""
    hooks: dict[str, list[dict[str, Any]]] = Field(default_factory=dict)
    global_scope: bool = False


class HookUpdate(BaseModel):
    """Model for updating a hook."""

    description: str | None = None
    hooks: dict[str, list[dict[str, Any]]] | None = None


class HookPreview(BaseModel):
    """Preview of merged hooks."""

    tags: list[str]
    preview: str
    settings: dict[str, Any]


class HookValidation(BaseModel):
    """Hook validation result."""

    tag: str
    valid: bool
    errors: list[str] = Field(default_factory=list)

File: dashboard/frontend/src/types/hook.ts
/**
 * Hook types for Claude Code hooks management
 */

export interface HookHandler {
  type: 'command' | 'prompt' | 'agent';
  command?: string;
  prompt?: string;
  timeout?: number;
  async?: boolean;
}

export interface HookMatcher {
  matcher?: string;
  hooks: HookHandler[];
}

export type HookEvent =
  | 'PreToolUse'
  | 'PostToolUse'
  | 'PostToolUseFailure'
  | 'Stop'
  | 'SessionStart'
  | 'SessionEnd'
  | 'Notification'
  | 'SubagentStart'
  | 'SubagentStop'
  | 'UserPromptSubmit'
  | 'PermissionRequest'
  | 'PreCompact';

export interface HookSummary {
  tag: string;
  description: string;
  location: 'project' | 'global';
  events: HookEvent[];
}

export interface Hook {
  tag: string;
  description: string;
  location: 'project' | 'global';
  hooks: Record<HookEvent, HookMatcher[]>;
}

export interface HookCreate {
  tag: string;
  description?: string;
  hooks?: Record<string, HookMatcher[]>;
  global_scope?: boolean;
}

export interface HookUpdate {
  description?: string;
  hooks?: Record<string, HookMatcher[]>;
}

export interface HookPreview {
  tags: string[];
  settings_json: string;
  events: HookEvent[];
}

export interface HookValidation {
  tag: string;
  valid: boolean;
  errors: string[];
}
